src/animation/state_machine_input.cpp

```diff
@@ -14,6 +14,7 @@ StatusCode StateMachineInput::import(ImportStack& importStack) {
     if (stateMachineImporter == nullptr) {
         return StatusCode::MissingObject;
     }
-    stateMachineImporter->addInput(this);
+    // WOW -- we're handing off ownership of this!
+    stateMachineImporter->addInput(std::unique_ptr<StateMachineInput>(this));
```

Comment:
😬

How many places is this called from? Can import() be converted to a static method that takes an std::unique_ptr<StateMachineInput> argument?

Reply:
Not sure. If we want to explore that refactor, it would be separate from this change I think.

Reply:
Fair enough